.travis.yml

```diff
@@ -2,26 +2,35 @@ language: php
 
 sudo: false
 
-php:
-  - 5.4
-  - 5.5
-  - 5.6
-  - 7.0
+matrix:
+    include:
+        - php: 5.4
+          env: test-exclude-groups=datetimeinterface
+        - php: 5.5
+        - php: 5.6
+        - php: 7.0
+        - php: 7.1
+        - php: 7.2
 
 services: mongodb
 
 before_install:
-  - if [[ "$TRAVIS_PHP_VERSION" = 5.* ]]; then echo 'extension=mongo.so' >> ~/.phpenv/versions/$(phpenv version-name)/etc/conf.d/travis.ini; fi
-  - if [[ "$TRAVIS_PHP_VERSION" != 5.* ]]; then echo 'extension=mongodb.so' >> ~/.phpenv/versions/$(phpenv version-name)/etc/conf.d/travis.ini; fi
-  - if [[ "$TRAVIS_PHP_VERSION" != 5.* ]]; then cp composer7.json composer.json; fi
+    - if [[ "$TRAVIS_PHP_VERSION" = 5.* ]]; then echo 'extension=mongo.so' >> ~/.phpenv/versions/$(phpenv version-name)/etc/conf.d/travis.ini; fi
+    - if [[ "$TRAVIS_PHP_VERSION" != 5.* ]]; then echo 'extension=mongodb.so' >> ~/.phpenv/versions/$(phpenv version-name)/etc/conf.d/travis.ini; fi
+    - if [[ "$TRAVIS_PHP_VERSION" != 5.* ]]; then cp composer7.json composer.json; fi
 
 install:
-  - composer install --prefer-dist
+    - composer install --prefer-dist
 
 script:
-  - bin/phpunit -c tests/
+    - |
+        if [[ ! $test-exclude-groups ]]; then
```

Comment:
try `if [[ ! -z "$test-exclude-groups" ]]; then`

Reply:
I'm pretty sure the problem isn't in the condition check, since the `else` condition is being executed for php 5.4. I'll push a commit proving this ASAP.

Reply:
I think `\DateTimeInterface::class` syntax is not supported in php 5.4

Reply:
because php fails simply with parse error, use full class name as string and check whether is sublass or whatever ;)

Reply:
Yes, it isn't supported. But this test is intended to be run with a version of PHP with support for `\DateTimeInterface`, so `::class` resolver must be available. Do you think we should change the test group name to something pointing to PHP version ("php55-or-greater" by instance) instead?

Reply:
I see with the latest commit that the executed command is `Debug: 'bin/phpunit -c tests/ --exclude-group -exclude-groups'`. I really don't know why Travis is stripping the "test" string from `$test-exclude-groups` environment variable and using the name instead of resolving it AS IS. I'll try to use another name for this variable in order to check if the result changes.

Reply:
use `${phpunit-exclude-groups}` a dash in variable name is not something you should usually use, since it works as a separator, better use underscores, dealing with bash is not that simple ;)

Reply:
but better use underscores instead of dash, will prevent more issues

Reply:
Great @l3pp4rd! Thank you. Good to know, issue seems to be solved. I'll squash my commits now.